paginator/paginator.go

```diff
@@ -120,9 +120,20 @@ func (p *Paginator) validate(dest interface{}) (err error) {
 func (p *Paginator) setup(db *gorm.DB, dest interface{}) {
 	var sqlTable string
 	for i := range p.rules {
+		multiKey := strings.Contains(p.rules[i].Key, ".")
 		if p.rules[i].SQLRepr == "" {
-			if sqlTable == "" {
-				sqlTable = db.NewScope(dest).TableName()
+			// if key has levels then recalculate table name regardless prev value
+			// because it can be different for different keys in an aggregated model
+			if sqlTable == "" || multiKey {
+				if multiKey {
+					subkeys := strings.Split(p.rules[i].Key, ".")
+					parentPath := strings.Join(subkeys[0:len(subkeys)-1], ".")
+					if parent, ok := util.ReflectFieldByPath(dest, parentPath); ok {
```

Comment:
Would it be cleaner to use the ReflectValueByPath

Reply:
Can't use `ReflectValueByPath` here because `dest` in this case is an empty slice and when it would try to value of an empty slice it panics so we have to work with Type here